components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+                var monValue = self._cronGenerator.find('#month').val();
+                if(isValidMonthValue(monValue)){
+                     self._cronGenerator.find('#month').removeClass("error-field");
+                     self._cronGenerator.find("#output").text(description);
+                     self._cronGenerator.find('#errorMsg').hide();
+                } else {
+                     self._cronGenerator.find('#month').addClass("error-field");
+                     self._cronGenerator.find('#errorMsg').text(errorMsg);
+                     self._cronGenerator.find('#errorMsg').show();
+                }
+                self.isErrorOccured();
+            });
+            self._cronGenerator.find("#dayWeek").on("input", function(){
+                 self._cronGenerator.find("#expression").val(self._cronGenerator.find("#second").val() + " " +
+                     self._cronGenerator.find("#minute").val()+ " " + self._cronGenerator.find("#hour").val() + " " +
+                     self._cronGenerator.find("#dayMonth").val() + " " +self._cronGenerator.find("#month").val() + " " +
+                     self._cronGenerator.find("#dayWeek").val());
+                 var expression = self._cronGenerator.find("#expression").val();
+                 var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+
+                 var weekValue = self._cronGenerator.find('#dayWeek').val();
+                 var dayMonValue = self._cronGenerator.find('#dayMonth').val();
+                 if(isValidDayOfWeekValue(weekValue, dayMonValue)){
+                      self._cronGenerator.find('#dayWeek').removeClass("error-field");
+                      self._cronGenerator.find("#output").text(description);
+                      self._cronGenerator.find('#errorMsg').hide();
+                 } else {
+                      self._cronGenerator.find('#dayWeek').addClass("error-field");
+                      self._cronGenerator.find('#errorMsg').text(errorMsg);
+                      self._cronGenerator.find('#errorMsg').show();
+                 }
+                 self.isErrorOccured();
+            });
+            self._cronGenerator.find("#expression").on("input",function(){
+                 var expression = self._cronGenerator.find("#expression").val();
+                 var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+                 var splitValues = expression.split(" ");
+                 var second = splitValues[0];
+                 var minute = splitValues[1];
+                 var hour = splitValues[2];
+                 var dayMonth = splitValues[3];
+                 var month = splitValues[4];
+                 var dayWeek = splitValues[5];
+                 if(isValidCronExpression(expression) === true){
```

Comment:
this is wrong way of coding. you can just check  isValidCronExpression() in the if block method name need to be changed.

Reply:
fixed in 0fb87f3